File: Guerby_Duval.Server/Models/Tarea.cs
using Guerby_Duval.Server.ModelValidations;
using System.ComponentModel.DataAnnotations;

namespace AngularApp3.Server.Models
{
    public class Tarea
    {
        public int Id { get; set; }

        [Required]
        public string Titulo { get; set; }

        [Required]
        public string Descripcion { get; set; }

        [Required]
        [TareaFecha]
        public DateTime FechaCreacion { get; set; }

        [Required]
        public DateTime FechaVencimiento { get; set; }

        public bool Completada { get; set; }
    }
}

File: guerby_duval.client/src/app/models/tarea.ts
export interface TareaInterface {
  id: number;
  titulo: string;
  descripcion: string;
  fechaCreacion: Date;
  fechaVencimiento: Date;
  completada: boolean;
}

export class Tarea implements TareaInterface {
  id: number;
  titulo: string;
  descripcion: string;
  fechaCreacion: Date;
  fechaVencimiento: Date;
  completada: boolean;

  constructor(
    id: number,
    titulo: string,
    descripcion: string,
    fechaCreacion: string,
    fechaVencimiento: string,
    completada: boolean
  ) {
    this.id = id;
    this.titulo = titulo;
    this.descripcion = descripcion;
    this.fechaCreacion = new Date(fechaCreacion);
    this.fechaVencimiento = new Date(fechaVencimiento);
    this.completada = completada;
  }
}
